File: [프로그래머스] 연습문제_올바른 괄호.py
# ( 가 나오면 스택에 넣기
# ) 가 나오면 스택에 (가 있는지 확인
# 스택이 비어있으면 true 리턴

def solution(s):
    stack = []
    for i in s:
        if i == "(":
            stack.append(i)
        else:
            if stack and stack.pop() == "(":
                continue
            else:
                return False
    return not stack    #스택이 비어있는 경우 true

File: [프로그래머스] 연습문제_올바른 괄호.js
function solution(s){
    let stack = [];
    
    for(let i of s){
        if (!stack){
            stack.push(i)
            continue;
        } 
        
        if(stack[stack.length-1] === "(" && i === ")"){
            stack.pop()
        } else {
            stack.push(i)
        }
    }

    return stack.length ? false : true ;
}